homeassistant/components/sharkiq/update_coordinator.py

```diff
@@ -76,9 +79,11 @@ async def _async_update_data(self) -> bool:
             SharkIqAuthExpiringError,
         ) as err:
             LOGGER.debug("Bad auth state.  Attempting re-auth", exc_info=err)
+            await self.ayla_api.async_sign_in()
             raise ConfigEntryAuthFailed from err
```

Comment:
I can see an issue here: The above `gather` has any of the `_async_update_vacuum` in the list of `online_vacs` throw an exception, then the error will arrive here. The state will potentially be partially updated across all the online_vacs.

You might want to use `return_exceptions=True` in the gather as [described here](https://docs.python.org/3/library/asyncio-task.html#running-tasks-concurrently) and retry only any ones that fail, and then throw an exception if they fail twice. You'll still have mixed state, but there's probs not much you can realistically do about that.

Also the exception just gets rethrown, rather than retry again from the top.